psi4/driver/procrouting/response/scf_response.py

```diff
@@ -92,132 +92,92 @@ def cpscf_linear_response(wfn, *args, **kwargs):
-                vector_names[i], shape[0], shape[1], nbf, nbf, ndocc, nvirt))
+    _print_header(complete_dict)
+
+    nbf = wfn.basisset().nbf()
+    Co = [wfn.Ca_subset("AO", "OCC"), wfn.Cb_subset("AO", "OCC")]
+    Cv = [wfn.Ca_subset("AO", "VIR"), wfn.Cb_subset("AO", "VIR")]
+    vectors_transformed = []
+    for vector in vectors:
+        if vector.shape != (nbf, nbf):
+            raise ValidationError(f"Vector must be of shape ({nbf}, {nbf}) for transformation"
+                                  " to the SO basis.")
+        v_a = core.triplet(Co[0], vector, Cv[0], True, False, False)
+        vectors_transformed.append(v_a)
+        if not restricted:
+            v_b = core.triplet(Co[1], vector, Cv[1], True, False, False)
+            vectors_transformed.append(v_b)
 
     # compute response vectors for each input vector
     params = [kwargs.pop("conv_tol", 1.e-5), kwargs.pop("max_iter", 10), kwargs.pop("print_lvl", 2)]
 
-    responses = wfn.cphf_solve(vectors, *params)
+    responses_list = wfn.cphf_solve(vectors_transformed, *params)
 
     # zip vectors, responses for easy access
-    vectors = {k: v for k, v in zip(vector_names, vectors)}
-    responses = {k: v for k, v in zip(vector_names, responses)}
-
+    if restricted:
+        vectors = {f"{k}_a": v for k, v in zip(vector_names, vectors_transformed)}
+        responses = {f"{k}_a": v for k, v in zip(vector_names, responses_list)}
+    else:
+        vectors = {f"{k}_a": v for k, v in zip(vector_names, vectors_transformed[::2])}
+        vectors.update({f"{k}_b": v for k, v in zip(vector_names, vectors_transformed[1::2])})
+        responses = {f"{k}_a": v for k, v in zip(vector_names, responses_list[::2])}
+        responses.update({f"{k}_b": v for k, v in zip(vector_names, responses_list[1::2])})
     # compute response values, format output
     output = []
+    pref = -4.0 if restricted else -2.0
     for prop in complete_dict:
-
-        # try to replicate the data structure of the input
-        if 'User' in prop['name']:
-            if prop['length'] == 0:
-                output.append(responses[prop['vector names'][0]])
-            else:
-                buf = []
-                for name in prop['vector names']:
-                    buf.append(responses[name])
-                output.append(buf)
-
-        else:
-            names = prop['vector names']
-            dim = len(names)
-
-            buf = np.zeros((dim, dim))
-
-            for i, i_name in enumerate(names):
-                for j, j_name in enumerate(names):
-                    buf[i, j] = -1.0 * vectors[i_name].vector_dot(responses[j_name])
-
-            output.append(buf)
+        names = prop['vector names']
+        dim = len(names)
+        buf = np.zeros((dim, dim))
+        for i, i_name in enumerate(names):
+            for j, j_name in enumerate(names):
+                buf[i, j] = pref * vectors[f"{i_name}_a"].vector_dot(responses[f"{j_name}_a"])
```

Comment:
I don't understand why `pref` changed from -1.0 to -2.0.

Reply:
I removed the prefactor -2.0 from the RHS in the RHF case, which was implicitly yielding another factor of 2.0 for the solution from CPHF... now handling of the prefactors is consistent, i.e., the prefactor did not change, it is just explicitly written out in one spot now.

Reply:
ah okay! Now I see it, thanks.